File: students/mauricio_acosta/lesson1/uva/the_new_version_of_a_game/game.go
//version del programa en go usando expresiones regulares para la validacion
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	for {
		t, err1 := reader.ReadString('\n') //al llamar a ReadString, se incluye el caracter '\n'
		w, err2 := reader.ReadString('\n')
		if err1 != nil || err2 != nil { // err1 == io.EOF || err2 == io.EOF
			break
		}
		w = strings.Replace(w, "?", ".", -1)
		fmt.Println(logic(t[:len(t)-1], w[:len(w)-1])) // se elimina el caracter '\n'
	}
}

func logic(t, w string) (count int) {
	for i := 0; i < len(t)-len(w)+1; i++ {
		if ok, _ := regexp.MatchString(w, t[i:i+len(w)]); ok {
			count++
		}
	}
	return
}

File: students/mauricio_acosta/lesson1/uva/the_new_version_of_a_game/game.py
# version del programa en python usando expresiones regulares para la validacion
import re
from sys import stdin

def logic(t,w):
    count = 0
    for i in range(len(t)-len(w)+1):
        a = t[i:i+len(w)]

        if re.match(w,a):
            count += 1
    return count

while True:
    t = stdin.readline() #al llamar a readline, se incluye el caracter '\n'
    w = stdin.readline()
    if not t or not w:
        break
    w = str.replace(w,"?",".")
    print(logic(t[:-1],w[:-1])) # se quita \n
